lib/Dialect/AIEVec/Transforms/ConvertVectorToAIEVec.cpp

```diff
@@ -1711,17 +1710,13 @@ configureAIEVecV2TransformationLegalizations(ConversionTarget &target) {
 }
 
 static void
-configureAIEVecConvOpTransformationLegalizations(ConversionTarget &target) {
+configureAIEVecConvOpTransformationLegalizations(ConversionTarget &target,
+                                                 AnalysisManager &am) {
   target.addLegalDialect<xilinx::aievec::AIEVecDialect>();
   target.addLegalDialect<arith::ArithDialect>();
-  target.addDynamicallyLegalOp<arith::AddIOp>([](arith::AddIOp op) {
-    SmallVector<SmallVector<arith::MulIOp, 8>, 8> groupFusedOps;
-    MulDefMapTy macChainMap;
-    unsigned dupFactor = 1;
-    bool hasMulConv = false;
-    Value acc = nullptr;
-    return !canFoldMulAddChainToConvOp(op, macChainMap, groupFusedOps,
-                                       dupFactor, hasMulConv, acc);
+  target.addDynamicallyLegalOp<arith::AddIOp>([&am](arith::AddIOp op) {
+    return !am.getChildAnalysis<canFoldMulAddChainToConvOpAnalysis>(op)
+                .canFoldMulAddChainToConvOp;
```

Comment:
Same here, `configure<Legalizations>` should be declare in the header file and implemented in the corresponding source file.